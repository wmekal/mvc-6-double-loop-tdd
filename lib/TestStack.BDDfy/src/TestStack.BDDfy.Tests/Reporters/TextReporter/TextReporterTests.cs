using System.Collections.Generic;
using System.Runtime.CompilerServices;
using System.Text;
using ApprovalTests;
using ApprovalTests.Utilities;
using TestStack.BDDfy.Reporters;
using Xunit;

namespace TestStack.BDDfy.Tests.Reporters.MarkDown
{
    public class TextReporterTests
    {
        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedReport()
        {
            var stories = new ReportTestData().CreateMixContainingEachTypeOfOutcomeWithOneScenarioPerStory();
            var actual = new StringBuilder();

            foreach (var story in stories)
            {
                var textReporter = new TextReporter();
                textReporter.Process(story);
                actual.AppendLine(textReporter.ToString());
            }

            Approvals.Verify(actual.ToString(), StackTraceScrubber.ScrubStackTrace);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedTextWithExamples()
        {
            var stories = new ReportTestData().CreateTwoStoriesEachWithOneFailingScenarioAndOnePassingScenarioWithThreeStepsOfFiveMillisecondsAndEachHasTwoExamples();
            var actual = new StringBuilder();

            foreach (var story in stories)
            {
                var textReporter = new TextReporter();
                textReporter.Process(story);
                actual.AppendLine(textReporter.ToString());
            }

            Approvals.Verify(actual.ToString(), StackTraceScrubber.ScrubStackTrace);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void LongStepName()
        {
            var textReporter = new TextReporter();
            var scenario = new Scenario(typeof(TextReporterTests), new List<Step>
            {
                new Step(o => null, new StepTitle("Given a normal length title"), false, ExecutionOrder.SetupState, true, new List<StepArgument>()),
                new Step(o => null, new StepTitle("When something of normal length happens"), false, ExecutionOrder.Transition, true, new List<StepArgument>()),
                new Step(o => null, new StepTitle("Then some long state should be: #Title\r\n\r\nSome more stuff which is quite long on the second line\r\n\r\nAnd finally another really long line"),
                    true, ExecutionOrder.Assertion, true, new List<StepArgument>()),
                new Step(o => null, new StepTitle("And a normal length assertion"), true, ExecutionOrder.ConsecutiveAssertion, true, new List<StepArgument>())
            }, "Scenario Text", new List<string>());
            textReporter.Process(new Story(new StoryMetadata(typeof(TextReporterTests), new StoryNarrativeAttribute()),
                scenario));
            var actual = new StringBuilder();
            actual.AppendLine(textReporter.ToString());
            Approvals.Verify(actual.ToString(), StackTraceScrubber.ScrubStackTrace);
        }
    }
}
