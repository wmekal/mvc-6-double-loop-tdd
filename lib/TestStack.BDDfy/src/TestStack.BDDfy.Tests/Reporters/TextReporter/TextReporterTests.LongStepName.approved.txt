Story: Text reporter tests

Scenario: Scenario Text
	Given a normal length title                                  [Not executed] 
	When something of normal length happens                      [Not executed] 
	Then some long state should be: #Title                       [Not executed] 

Some more stuff which is quite long on the second line

And finally another really long line
	  And a normal length assertion                              [Not executed] 


